# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/xawtv/hotfix-sys_siglist.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

console/record.c: In function 'ctrlc':
console/record.c:432:17: error: 'sys_siglist' undeclared (first use in this function)
  432 |                 sys_siglist[signal]);

--- xawtv-3.107/console/record.c.vanilla	2025-06-24 17:41:02.967405093 +0200
+++ xawtv-3.107/console/record.c	2025-06-24 17:41:21.199768888 +0200
@@ -429,7 +429,7 @@
 {
     if (verbose)
 	fprintf(stderr,"\n%s - exiting\n",
-		sys_siglist[signal]);
+		strsignal(signal));
     stop = 1;
 }

console/fbtools.c: In function 'fb_catch_exit_signals':
console/fbtools.c:523:33: error: 'sys_siglist' undeclared (first use in this function)
  523 |     fprintf(stderr,"Oops: %s\n",sys_siglist[termsig]);
 
--- xawtv-3.107/console/fbtools.c.vanilla	2025-06-24 17:46:33.825859550 +0200
+++ xawtv-3.107/console/fbtools.c	2025-06-24 17:46:59.133392040 +0200
@@ -520,6 +520,6 @@
 
     /* cleanup */
     fb_cleanup();
-    fprintf(stderr,"Oops: %s\n",sys_siglist[termsig]);
+    fprintf(stderr,"Oops: %s\n", strsignal(termsig));
     exit(42);
 }
